bin/generate_inputs.py

```diff
@@ -83,54 +81,86 @@ def load_xml(assembly):
             return None
 
 
-def write_samplesheet_line(assembly, assembly_path, run, run_path, samplesheet):
-    with open(samplesheet, 'a') as file_out:
-        if assembly_path:
-            # TODO: return s3 when nextflow will work with -resume
-            transformed_assembly, _ = transform_paths(assembly_path)
-        else:
-            print(f'no assembly path {assembly}')
-        chosen_run_path = []
-        if run_path:
-            se, pe_forward, pe_reversed = False, False, False
-            if len(run_path) >= 2:
-                for raw_file in run_path:
-                    if len(raw_file.split('_1')) >= 2:
-                        pe_forward = raw_file
-                    elif len(raw_file.split('_2')) >= 2:
-                        pe_reversed = raw_file
-                    else:
-                        se = raw_file
-                if se and pe_forward and pe_reversed:
-                    print(f"More than 3 raw files detected for {run}, choosing PE reads")
-                    chosen_run_path = [pe_forward, pe_reversed]
-                elif pe_forward and pe_reversed:
-                    chosen_run_path = [pe_forward, pe_reversed]
+def generate_software_input(erz_acc):
+    handler_request = handler.get_assembly(erz_acc)
+    assembly_software = handler_request["assembly_software"]
+    if not assembly_software:
+        try:
+            json_analysis = load_xml(erz_acc)
+            assembly_software = json_analysis["ANALYSIS_SET"]["ANALYSIS"]["ANALYSIS_TYPE"]["SEQUENCE_ASSEMBLY"][
+                "PROGRAM"]
+            assembly_software = assembly_software.replace(" ", "_" if "v" in assembly_software else "_v")
+            if '_v' not in assembly_software:
+                assembly_software = assembly_software.replace('v', '_v')
+        except:
+            print('Did not get software from XML. Leaving empty field')
```

Comment:
```suggestion
            print('Could not get the assembly software from ENA API. Leaving the field empty')
```
Just to clarify what attempt failed